gcsio/src/test/java/com/google/cloud/hadoop/gcsio/integration/GoogleCloudStorageImplTest.java

```diff
@@ -209,6 +209,62 @@ public void testCopySingleItemWithRewrite() throws IOException {
     assertObjectContent(gcs, copiedResourceId, partition, partitionsCount);
   }
 
+  @Test
+  public void testUploadAndGetObjectWithEncryptionKey() throws IOException {
```

Comment:
May you extract these tests in a separate `CmekEncryptionTest` class - we have a bad habit to put all test in a single class, but it's getting harder and harder to maintain.

Reply:
Done.